source/MaterialXGenGlsl/GlslShaderGenerator.h

```diff
@@ -52,6 +52,10 @@ class GlslShaderGenerator : public HwShaderGenerator
     /// the shader generator. The enumeration may be converted to a different type than the input.
     bool remapEnumeration(const ValueElement& input, const string& value, std::pair<const TypeDesc*, ValuePtr>& result) const override;
 
+
+    /// Helper to emit active light count definition
+    static void emitActiveLightSourceDefinition(GenContext& context, ShaderStage& stage);
```

Comment:
You should not require a static, since GenContext has a shaderGenerator data member. It would need to be cast the correct type. Also seems this should belong on HwShaderGenerator.h so all hardware shader generators can make use of this.